### PR TITLE
Graduate NavigationList from github.com to primer/view_components

### PR DESCRIPTION
**Pull request by camertron**
We've been using `NavigationList` for settings pages on github.com for a while now and are ready to graduate the component into primer/view_components.

### CODE REVIEW

**Comment by camertron** [comment]
Oh shoot I forgot about the javascript! Coming soon.

**Review by jonrohan — APPROVED** [review]
Minus the catalyst/not-catalyst question, this looks good 👍🏻

File: tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "compilerOptions": {
     "target": "es2017",
+    "experimentalDecorators": true,
```
Comment (keithamus) [inline tsconfig.json:4]: This shouldn't be needed I think.

**Review by hectahertz — APPROVED** [review]
Looking good! 👍

File: app/components/primer/alpha/navigation_list_element.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        item.setAttribute('aria-expanded', 'true')
+      }
+    }
+
+    e.stopPropagation()
+  }
+}
+
+export {}
+
```
Comment (keithamus) [inline app/components/primer/alpha/navigation_list_element.ts:29]: Is this line necessary? It might be better to instead export the class on L1

**Pull request closed without merging**